src/lighteval/metrics/__init__.py

```diff
@@ -116,7 +133,7 @@ def apply_generative_metric(
 
 def apply_multichoice_metric(results: list[ModelReturn], formatted_doc: Doc, metrics: list[Metric]):
     outputs = {}
-    mc_results = results[: len(formatted_doc.choices)]
+    mc_results = [results.pop(0) for _ in range(len(formatted_doc.choices))]
```

Comment:
@clefourrier , return statement shouldn't become the following, then? I.e.

```python
return results[len(formatted_doc.choices) :], outputs
```
to

```python
return results, outputs
```

Reply:
True, since now we do a pop.
Good catch! Can you make a review suggestion so I can give you authorship?

Reply:
```suggestion
    mc_results = [results.pop(0) for _ in range(len(formatted_doc.choices))]
```

Reply:
changed